py4j-java/src/test/java/py4j/reflection/ReflectionEngineTest.java

```diff
@@ -202,6 +203,87 @@ public void testGetMethod() {
+		System.out.println("Number.class.isAssignable : " + Number.class.isAssignableFrom(Integer.class));
+		System.out.println("Double.class.isAssignable : " + Double.class.isAssignableFrom(Integer.class));
+		System.out.println("Integer.class.isAssignable: " + Integer.class.isAssignableFrom(Double.class));
+		System.out.println("Number.class.isAssignable : " + Number.class.isAssignableFrom(int.class));
+		System.out.println("Integer.class.isAssignable: " + Integer.class.isAssignableFrom(int.class));
+		System.out.println("int.class.isAssignable    : " + int.class.isAssignableFrom(Integer.class));
+	}
+
+	@Test
+	public void testVarArgsConstructor() {
+		ReflectionEngine engine = new ReflectionEngine();
+		MethodInvoker mInvoker = engine.getConstructor("p1.VarArgTester", new Object[] {});
+		assertArrayEquals(new Class[] {int[].class}, mInvoker.getConstructor().getParameterTypes());
+		VarArgTester tester = (VarArgTester) mInvoker.invoke(null);
+		assertEquals(1, tester.called());
+
+		mInvoker = engine.getConstructor("p1.VarArgTester", 1);
+		assertArrayEquals(new Class[] {int.class}, mInvoker.getConstructor().getParameterTypes());
+		tester = (VarArgTester) mInvoker.invoke(null, 1);
+		assertEquals(0, tester.called());
+
+		mInvoker = engine.getConstructor("p1.VarArgTester", new Object[] {1, 2});
+		assertArrayEquals(new Class[] {int[].class}, mInvoker.getConstructor().getParameterTypes());
+		tester = (VarArgTester) mInvoker.invoke(null, 1, 2);
+		assertEquals(1, tester.called());
+
+		mInvoker = engine.getConstructor("p1.VarArgTester", new Object[] {new int[] {1, 2}});
+		assertArrayEquals(new Class[] {int[].class}, mInvoker.getConstructor().getParameterTypes());
+		tester = (VarArgTester) mInvoker.invoke(null, new int[] {1, 2});
+		assertEquals(1, tester.called());
+
+		mInvoker = engine.getConstructor("p1.VarArgTester", new int[] {1, 2});
+		assertArrayEquals(new Class[] {int[].class}, mInvoker.getConstructor().getParameterTypes());
+		tester = (VarArgTester) mInvoker.invoke(null, new int[] {1, 2});
+		assertEquals(1, tester.called());
+
+		mInvoker = engine.getConstructor("p1.VarArgTester", new Object[] {1.5f});
+		assertArrayEquals(new Class[] {float.class, int[].class}, mInvoker.getConstructor().getParameterTypes());
+		tester = (VarArgTester) mInvoker.invoke(null, 1.5f);
+		assertEquals(2, tester.called());
+	}
+
+	@Test
+	public void testVarArgsMethod() {
+		ReflectionEngine engine = new ReflectionEngine();
+		VarArgTester tester = new VarArgTester();
+		MethodInvoker mInvoker;
+
+		mInvoker = engine.getMethod(tester, "method1", new Object[] {}); // gets #method1()
+		assertArrayEquals(new Class[] {}, mInvoker.getMethod().getParameterTypes());
+		assertEquals(tester.method1(), mInvoker.invoke(tester));
+
+		mInvoker = engine.getMethod(tester, "method1", (Object[]) null); // gets #method1()
+//		assertArrayEquals(new Class[] {int[].class}, mInvoker.getMethod().getParameterTypes());
```

Comment:
As I was developing the method, the result changed as desired. I will remove the line.